kubemarine/procedures/haproxy_mntc.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2021-2023 NetCracker Technology Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from collections import OrderedDict
+import time
+from kubemarine.core import flow
+from kubemarine.core.action import Action
+from kubemarine.core.resources import DynamicResources
+from kubemarine.haproxy import is_maintenance_mode
+
+def haproxy_mntc(cluster):
```

Comment:
I suggest another approach, which is more complex but more Kubemarine-like in my opinion.
1. Introduce new inventory
   ```yaml
   services:
     loadbalancer:
       haproxy:
         maintenance_mode:
           install: False
           enabled: False
           config_location: /etc/haproxy/haproxy_mntc.cfg
           keep_updated: True
   ```
   The above are default values (defaults.yaml)
   install - whether to install maintenance config haproxy_mntc.cfg. Has the same meaning as the `services.loadbalancer.haproxy.maintenance_mode` currently.
   enabled - whether to enable the maintenance mode on balancer nodes.
   config_location - moved from `services.loadbalancer.haproxy.mntc_config_location`
   keep_updated - If false, `kubemarine install --tasks deploy.loadbalancer.haproxy.configure` will not switch mode (see p. 2 below).
2. `install --tasks deploy.loadbalancer.haproxy.configure` configures and installs both configs if `services.loadbalancer.haproxy.keep_configs_updated == True` and switches mode if `maintenance_mode.keep_updated == True`
3. add/remove procedures only configure and install both configs if `services.loadbalancer.haproxy.keep_configs_updated == True`
4. New `haproxy_mntc` is not necessary in this approach, but if implemented, it only switches mode ignoring value of `maintenance_mode.keep_updated`. It has procedure inventory of the similar format as in p. 1 but supporting only `maintenance_mode.enabled`.
5. Write patch (see [Kubemarine Migration Procedure](https://github.com/Netcracker/KubeMarine/blob/main/documentation/Maintenance.md#kubemarine-migration-procedure) that changes format of `haproxy.maintenance_mode` in inventory, and sets `maintenance_mode.keep_updated=False` for backward compatibility.

Reply:
Hi,
Yes, the suggested approach is more complex.
But the approach used in this PR is simpler and more effective as well.
We are able to enable/disable maintenance mode on haproxy with ease. 
If maintenance mode is not supported by the cluster this procedure will not work in that cluster and will give the appropriate error message.

Reply:
Let's have an arch discussion about this topic. Until then - postponed.

Reply:
Decided to cancel current implementation in the favor of provided above proposal. The further implementation should be start with detailed ADR and based on this comment